### PR TITLE
Add Go SDK extension

### LINKED ISSUE #210: SDK for compiling Go programs
Is there any (planned) SDK for compiling Go programs for Flathub?
(I would like to try to package a program that uses libraries from [go-gl](https://github.com/go-gl) )

### PR DESCRIPTION
**Pull request by Barthalion**
Fixes #210

### CODE REVIEW

**Comment by Barthalion** [comment]
bot, build org.freedesktop.Sdk.Extension.golang

**Comment by nedrichards** [comment]
If we want this available at runtime for apps you're going to need an install.sh to call as well, like the JDK: https://github.com/flathub/org.freedesktop.Sdk.Extension.openjdk9/blob/master/org.freedesktop.Sdk.Extension.openjdk9.json#L168

**Comment by Barthalion** [comment]
Hm, no opinion about making it installable, I'm fine either way.

File: org.freedesktop.Sdk.Extension.golang.json [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+          ],
+          "url": "https://dl.google.com/go/go1.9.2.linux-386.tar.gz",
+          "sha256": "574b2c4b1a248e58ef7d1f825beda15429610a2316d9cbd3096d8d3fa8c0bc1a"
+        },
+        {
+          "type": "archive",
+          "only-arches": [
+            "arm"
+          ],
+          "url": "https://dl.google.com/go/go1.9.2.linux-armv6l.tar.gz",
```
Comment (nedrichards) [inline org.freedesktop.Sdk.Extension.golang.json]: I'm pretty sure this is the wrong arm architecture for our definition of ARM. We should test, but we may have to ban arm from this extension

Comment (Barthalion) [reply org.freedesktop.Sdk.Extension.golang.json]: Ah, right, we're targetting armv7? I could probably make some workaround to build it from source, I'll take a look at it.

Comment (TingPing) [reply org.freedesktop.Sdk.Extension.golang.json]: If we can just build it from source that would be nice in general.

Comment (Barthalion) [reply org.freedesktop.Sdk.Extension.golang.json]: After another look, seems to be potential time sink. Last release that doesn't require working Go compiler is 1.4 and it doesn't seem to build on my Odroid C2 (aarch64).

One option would be another extension to build this one (that would also cut some time for building), but use binary release just for aarch64.

Comment (nedrichards) [reply org.freedesktop.Sdk.Extension.golang.json]: Yeah - I'd rather have go, and the bootstrap is super messy.

Comment (nedrichards) [reply org.freedesktop.Sdk.Extension.golang.json]: @ramcq which exact arm do we use again? you were the last person who touched that as I remember ;-)

Comment (Barthalion) [reply org.freedesktop.Sdk.Extension.golang.json]: Looking at how gcc is built, it's armv7-a. Fix will come in a minute…

Comment (Barthalion) [reply org.freedesktop.Sdk.Extension.golang.json]: My bad, there are no binaries optimized for armv7. I will test armv6 one on my Odroid XU4 and merge it as is if it works properly.

**Comment by Barthalion** [comment]
bot, build org.freedesktop.Sdk.Extension.golang

**Pull request closed without merging**